content/spin/sqlite-api-guide.md

```diff
@@ -0,0 +1,184 @@
+
+To use SQLite functions, use the `spin_sqlite` module in the Python SDK. The `sqlite_open` and `sqlite_open_default` functions return a connection object. The connection object provides the `execute` method as described above. For example:
+
+```python
+from spin_http import Response
+from spin_sqlite import sqlite_open_default
+
+def handle_request(request):
+    conn = sqlite_open_default()
+    result = conn.execute("SELECT * FROM todos WHERE id > (?);", [1])
+    rows = result.rows()
+    return Response(200,
+                    {"content-type": "application/json"},
+                    bytes(str(rows), "utf-8"))
+```
+
+**General Notes**
+* Parameters are Python values (numbers, strings, and lists). Spin infers the underlying SQL type.
+* The `execute` method returns an object with `rows` and `columns` methods. `columns` returns a list of strings representing column names. `rows` is an array of rows, each of which is an array of Python values (as above) in the same order as `columns`.
+* The connection object doesn't surface the `close` function.
+* Errors are surfaced as exceptions.
+
+{{ blockEnd }}
+
+{{ startTab "TinyGo"}}
+
+The Go SDK doesn't currently surface the SQLite API.
+
+{{ blockEnd }}
+
+{{ blockEnd }}
+
+## Preparing an SQLite Database
+
+Although Spin provides SQLite as a built-in database, SQLite still needs you to create its tables.  Although you can do this using Spin components and the SQLite API, this usually results in database setup getting uncomfortably mixed in with application logic.
+
+Instead, you can use the `spin up --sqlite` option to run whatever SQL statements you need before your application starts.  This is typically used to create or alter tables, but can be used for whatever other maintenance or troubleshooting tasks you need.
+
+You can run a SQL script from a file using the `@filename` syntax:
+
+<!-- @nocpy -->
+
+```bash
+spin up --sqlite @migration.sql
+```
+
+Or you can pass SQL statements directly on the command line as a (quoted) string:
+
+<!-- @nocpy -->
+
+```bash
+spin up --sqlite "CREATE TABLE IF NOT EXISTS todos (id INTEGER PRIMARY KEY AUTOINCREMENT, description TEXT NOT NULL, due TEXT NOT NULL)"
+```
+
+You can provide the `--sqlite` flag more than once; Spin runs the statements in the order you provide them.
+
+## Custom SQLite Databases
+
+Spin defines a database named `"default"` and provides automatic backing storage.  If you need to customize Spin with additional databases, or to change the backing storage for the default database, you can do so via the `--runtime-config-file` flag and the `runtime-config.toml` file.  See [SQLite Database Runtime Configuration](/spin/dynamic-configuration#sqlite-storage-runtime-configuration) for details.
+
+## Granting SQLite Database Permissions to Components
+
+By default, a given component of an app will not have access to any SQLite databases. Access must be granted specifically to each component via the component manifest:
+
+```toml
```

Comment:
It may be useful to add multiple components to this example, and only grant 1 access to flush out the concept a bit more for the readers.